cmd/podman/farm/remove.go

```diff
@@ -43,57 +43,55 @@ func init() {
 }
 
 func rm(cmd *cobra.Command, args []string) error {
-	cfg, err := config.ReadCustomConfig()
-	if err != nil {
-		return err
-	}
-
-	if rmOpts.All {
-		cfg.Farms.List = make(map[string][]string)
-		cfg.Farms.Default = ""
-		if err := cfg.Write(); err != nil {
-			return err
+	deletedFarms := []string{}
+	err := config.EditConnectionConfig(func(cfg *config.ConnectionsFile) error {
+		if rmOpts.All {
+			cfg.Farm.List = make(map[string][]string)
+			cfg.Farm.Default = ""
+			return nil
 		}
-		fmt.Println("All farms have been deleted")
-		return nil
-	}
 
-	// If the --all is not set, we require at least one arg
-	if len(args) == 0 {
-		return errors.New("requires at lease 1 arg(s), received 0")
-	}
+		// If the --all is not set, we require at least one arg
+		if len(args) == 0 {
+			return errors.New("requires at lease 1 arg(s), received 0")
+		}
 
-	if len(cfg.Farms.List) == 0 {
-		return errors.New("no existing farms; nothing to remove")
-	}
+		if len(cfg.Farm.List) == 0 {
+			return errors.New("no existing farms; nothing to remove")
+		}
 
-	deletedFarms := []string{}
-	for _, k := range args {
-		if _, ok := cfg.Farms.List[k]; !ok {
-			logrus.Warnf("farm %q doesn't exist; nothing to remove", k)
-			continue
+		for _, k := range args {
+			if _, ok := cfg.Farm.List[k]; !ok {
+				logrus.Warnf("farm %q doesn't exist; nothing to remove", k)
+				continue
+			}
+			delete(cfg.Farm.List, k)
+			deletedFarms = append(deletedFarms, k)
+			if k == cfg.Farm.Default {
+				cfg.Farm.Default = ""
+			}
 		}
-		delete(cfg.Farms.List, k)
-		deletedFarms = append(deletedFarms, k)
-		if k == cfg.Farms.Default {
-			cfg.Farms.Default = ""
+		// Return error if none of the given farms were deleted
+		if len(deletedFarms) == 0 {
+			return fmt.Errorf("failed to delete farms %q", args)
 		}
-	}
-	// Return error if none of the given farms were deleted
-	if len(deletedFarms) == 0 {
-		return fmt.Errorf("failed to delete farms %q", args)
-	}
 
-	// Set a new default farm if the current default farm has been removed
-	if cfg.Farms.Default == "" && cfg.Farms.List != nil {
-		for k := range cfg.Farms.List {
-			cfg.Farms.Default = k
-			break
+		// Set a new default farm if the current default farm has been removed
+		if cfg.Farm.Default == "" && cfg.Farm.List != nil {
+			for k := range cfg.Farm.List {
+				cfg.Farm.Default = k
```

Comment:
not required for this pr: i don't care for this behavior and i see it was already there before you ... i think this is unexpected if i am a user.  in system connections, i think we just don't set a default (i could be very wrong on that).  in any case, it might be a card and conversation.